backend/server/src/db/model/build.rs

```diff
@@ -401,3 +325,115 @@ pub struct DrvRefs {
     /// Also known as dependency.
     pub reference: DrvId,
 }
+
+mod state {
```

Comment:
Is this module really necessary? It's a little odd to not put the trait implementations for `DrvBuildState` in the same module it's defined in

Reply:
looks like this was mostly done to avoid polluting the model namespace. Since it's mostly there to define traits, I think this is fine. I'm not well versed enough in rust to have a strong opinion.

Reply:
if this becomes awkward later, we can also revisit it.